decidim-forms/app/views/decidim/forms/questionnaires/show.html.erb

```diff
@@ -42,30 +42,44 @@
 
                 <%= decidim_form_for(@form, url: update_url, method: :post, html: { class: "form answer-questionnaire" }) do |form| %>
                   <%= invisible_captcha %>
-                  <% @form.responses.each_with_index do |answer, answer_idx| %>
-                    <div class="row column answer" data-max-choices="<%= answer.question.max_choices %>">
-                      <%= fields_for "questionnaire[responses][#{answer_idx}]", answer do |answer_form| %>
-                        <%= render "decidim/forms/questionnaires/answer", answer_form: answer_form, answer: answer, answer_idx: answer_idx, disabled: !current_participatory_space.can_participate?(current_user) %>
-                      <% end %>
-                    </div>
-                  <% end %>
+                  <% answer_idx = 0 %>
+                  <% @form.responses_by_step.each_with_index do |step_answers, step_index| %>
+                    <h4 class="section-heading">
+                      <%= t(".current_step", step: step_index + 1) %> <span class="answer-questionnaire__steps"><%= t(".of_total_steps", total_steps: @form.total_steps) %></span>
+                    </h4>
 
-                  <% if show_represent_user_group? %>
-                    <div class="row column represent-user-group">
-                      <%= cell("decidim/represent_user_group", form) %>
-                    </div>
-                  <% end %>
+                    <% step_answers.each do |answer| %>
```

Comment:
could you use `each_with_index` for the `answer_idx` variable?

Reply:
I don't think so, because I need to keep the counter between steps (which would be lost). Also, I need to skip the separators, so I'd need to keep track of those elements... I think it's easier to keep track using the `answer_ids`, but I'll give it another thought!